src/app/util/attribute-storage.cpp

```diff
@@ -336,6 +344,11 @@ CHIP_ERROR emberAfSetDynamicEndpoint(uint16_t index, EndpointId id, const EmberA
     emAfEndpoints[index].deviceTypeList = deviceTypeList;
     emAfEndpoints[index].endpointType   = ep;
     emAfEndpoints[index].dataVersions   = dataVersionStorage.data();
+#if CHIP_CONFIG_USE_ENDPOINT_UNIQUE_ID
+    MutableCharSpan targetSpan(emAfEndpoints[index].endpointUniqueId);
+    CopyCharSpanToMutableCharSpanWithTruncation(endpointUniqueId, targetSpan);
+    emAfEndpoints[index].endpointUniqueIdSize = static_cast<uint8_t>(targetSpan.size());
```

Comment:
Should static_assert that `sizeof(emAfEndpoints[index].endpointUniqueId) <= UINT8_MAX`, right?